### PR TITLE
Adding Expense model

### PR DESCRIPTION
**Pull request by FaezZangariya**
added Expense model and related migration

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -1,2 +1,5 @@
 [flake8]
-max-line-length = 120
+max-line-length = 300
+per-file-ignores =
+    # line too long
+    commune_app/migrations/0001_initial.py: E501
```
Comment (bennypowers) [inline .flake8:5]: can make this a glob to ignore all generated files

File: commune_app/models.py [hunk 2]
```diff
@@ -1,3 +1,63 @@
...
+from django.utils import timezone
+from django.contrib.auth.models import AbstractUser
+from django.core.validators import RegexValidator
 
-# Create your models here.
+
+phone_regex = RegexValidator(regex=r'^\+?1?\d{9,15}$', message="Phone number format: '+999999999'")
+
+
+class User(AbstractUser):
```
Comment (bennypowers) [inline commune_app/models.py:10]: we'd prefer to have docstrings on all classes

**Review by bennypowers — CHANGES_REQUESTED** [review]
small changes otherwise lgtm

File: commune_app/tests/test_Expense_model.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
+from ..models import Expense, User
+import pytest
+
+
+class TestExpenseModel:
+    @pytest.mark.django_db
+    def test_new_expense(self):
+        expense = Expense(title="buying some food", budget="150", date="2011-09-01T13:20:30+03:00", assign=User(first_name="Faez", last_name="Zangariya", email="faezzangariyacs@gmail.com", phone_number="+972523324193"))
+        assert expense.title == "buying some food"
+        assert expense.budget == "150"
```
Comment (kobihk) [inline commune_app/tests/test_Expense_model.py:10]: I think it should be a number(int) not a string
```suggestion
        assert expense.budget == 150
```

File: commune_app/tests/test_Expense_model.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    @pytest.mark.django_db
+    def test_create_expense(self):
+        expense = Expense.create_expense(title="Laundry", budget="200", date="2011-09-01T13:20:30+03:00", assign=User.create_user(first_name="Ron", last_name="Tur", email="rontur@gmail.com", phone_number="+972523324193"))
+        assert expense.title == "Laundry"
+        assert expense.budget == "200"
+        assert expense.date == "2011-09-01T13:20:30+03:00"
+        assert expense.assign.first_name == "Ron"
+
+    @pytest.mark.django_db
+    def test_get_expense_by_title(self):
```
Comment (kobihk) [inline commune_app/tests/test_Expense_model.py:23]: same as above 
you could use the same user you already created as a fixture like:
```suggestion
    def test_get_expense_by_title(self, user1):
```

File: commune_app/tests/test_Expense_model.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+        currU = User.create_user(first_name="ofir", last_name="tza", email="ofirtza@gmail.com", phone_number="+972523324193")
+        expense1 = Expense.create_expense(title="X", budget="100", date="2011-09-01T13:20:30+03:00", assign=currU)
+        expense2 = Expense.create_expense(title="Y", budget="200", date="2011-09-01T13:20:30+03:00", assign=currU)
+        expense3 = Expense.create_expense(title="Z", budget="300", date="2011-09-01T13:20:30+03:00", assign=currU)
```
Comment (kobihk) [inline commune_app/tests/test_Expense_model.py:27]: you can do the same as I suggested with user1 regarding the expenses to do them as a fixture and reuse them

**Review by kobihk — CHANGES_REQUESTED** [review]
@FaezZangariya this is the first PR I can see tests. 
great!!!
There is a code that is relevant to the user PR but I ignored it

File: commune_app/models.py [hunk 6]
```diff
@@ -1,3 +1,31 @@
...
+    id = models.IntegerField(unique=True, primary_key=True)
+    first_name = models.CharField(max_length=50)
+    last_name = models.CharField(max_length=50)
+    email = models.EmailField(max_length=250, unique=True)
+    phone_number = models.CharField(validators=[phone_regex], max_length=17, blank=True)
+
+
+class Expense(models.Model):
```
Comment (RonTuretzky) [inline commune_app/models.py:18]:
```suggestion
class Expense(models.Model):
    id = models.IntegerField(unique=True, primary_key=True)
```
adding id so you can use it for the getter later

File: commune_app/models.py [hunk 7]
```diff
@@ -1,3 +1,31 @@
...
+    title = models.CharField(max_length=50)
+    budget = models.IntegerField()
+    date = models.DateTimeField(default=timezone.now)
+    assign = models.ForeignKey(User, on_delete=models.CASCADE)
+
+    # returns an expense matching the title given as paramater
+    @staticmethod
+    def get_expense_by_title(title):
```
Comment (RonTuretzky) [inline commune_app/models.py:26]:
```suggestion
    def get_expense_by_pk(pk):
```

File: commune_app/models.py [hunk 8]
```diff
@@ -1,3 +1,31 @@
...
+        try:
+            expense = Expense.objects.filter(title=title).order_by('date').first()
```
Comment (RonTuretzky) [inline commune_app/models.py:28]:
```suggestion
            expense = Expense.objects.filter(id=id)
```

**Review by RonTuretzky — CHANGES_REQUESTED** [review]

**Review by ofirtzabari — APPROVED** [review]
Look good

**Comment by FaezZangariya** [comment]
We decided that the Expense model is not really needed

**Pull request closed without merging**